misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt

```diff
@@ -107,8 +106,13 @@ internal class SessionFactoryService(
 
     // NB: MetadataSources.getMetadataBuilder() returns a different instance each call!
     val metadataBuilder = metadataSources.metadataBuilder
+    // After `build` is called Hibernate creates ID Generators so we have to register this custom
+    // type before we call build, otherwise Id will be treated as a SerializableType instead.
+    // See: Component#createIdentifierGenerator
+    //metadataBuilder.applyBasicType(IdType(), Id::class.jvmName)
     val metadata = metadataBuilder.build() as MetadataImplementor
```

Comment:
would it be better to rename this to be like `draftMetadata` so the actual one gets the good name?

Reply:
Renamed